Added functionality

skipGit opened the pull request:
1. Added delete todo.
2. Added toggle completed todo.

skipGit committed:
Added functionality

1. Added delete todo.
2. Added toggle completed todo.

skipGit committed:
Added edit ToDo

Added edit ToDo functionality.

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/to-do-item.style.js:
```diff
@@ -13,4 +13,10 @@ export const useToDoItemClasses = makeStyles({
     paddingLeft: "20px",
     paddingTop: "10px",
   },
+  typo_isCompleted:{
```
use only camelCase for selectors

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/ToDoItem.jsx:
```diff
@@ -1,35 +1,85 @@
-        </CardContent>
-      </Card>
+      {isEditing ? (
+        <Card>
+          <FormControl>
+            <CardContent>
+              <Input
+                value={edited}
+                onChange={(event) => setEdited(event.target.value)}
+              ></Input>
+              <IconButton onClick={submitEdit}>
+                <CheckCircleSharpIcon></CheckCircleSharpIcon>
+              </IconButton>
+            </CardContent>
+          </FormControl>
+        </Card>
```
Create a component for this template. Modules should be small as possible

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/ToDoItem.jsx:
```diff
@@ -1,35 +1,85 @@
+          </FormControl>
+        </Card>
+      ) : (
+        <Card>
+          <CardContent className={classes.content}>
+            <Typography
+              className={isCompleted ? classes.typo_isCompleted : classes.typo}
+              align="left"
+              onClick={() => console.log(isEditing)}
+            >
+              {message}
+            </Typography>
+            <IconButton onClick={handleEdit}>
+              <EditIcon fontSize="small" />
+            </IconButton>
+            <IconButton onClick={handleComplete}>
+              <TaskIcon fontSize="small" />
+            </IconButton>
+            <IconButton onClick={handleDelete}>
+              <DeleteSharpIcon fontSize="small" />
+            </IconButton>
+          </CardContent>
+        </Card>
```
Create a component for this template. Modules should be small as possible

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/ToDoItem.jsx:
```diff
@@ -1,35 +1,85 @@
   const classes = useToDoItemClasses();
   const dispatch = useDispatch();
-  
+  const [isEditing, setIsEditing] = useState(false);
```
rename to editMode, setEditMode

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/ToDoItem.jsx:
```diff
@@ -1,35 +1,85 @@
 
-const ToDoItem = ({message, isCompleted}) => {
+const ToDoItem = ({ message, isCompleted, id }) => {
+  const [edited, setEdited] = useState(message);
```
rename to editedTitle, setEditedTitle

DmitryKirilenko16 commented on src/templates/to-do-list/to-do-item/ToDoItem.jsx:
```diff
@@ -1,35 +1,85 @@
+} from "../../../store/toDo/actions";
 
-const ToDoItem = ({message, isCompleted}) => {
+const ToDoItem = ({ message, isCompleted, id }) => {
```
rename message to title

skipGit committed:
Refactoring ToDoItem Component

1. Create EditToDoForm template.
2. Rename few values with appropriate names.

skipGit committed:
Create ToDocontent template

Added template for ToDoItem content render.

skipGit committed:
Fixed new item generation

Move logic of creating new ToDoItem to reducer.
Added id generator lib instead of using Date.now().
